log4j-jul/src/test/java/org/apache/logging/log4j/jul/test/CoreLoggerTest.java

```diff
@@ -100,6 +103,35 @@ public void testSetLevel() throws Exception {
         assertThat(childLogger.isLoggable(Level.ALL), is(false));
     }
 
+    @Test
+    public void testSetLevelIssue2281() {
+        final Logger a = Logger.getLogger("a");
+        final Logger a_b = Logger.getLogger("a.b");
+        final Logger a_b_c = Logger.getLogger("a.b.c");
+        // test default for this test
+        assertEquals(Level.INFO, a.getLevel());
+        assertEquals(Level.INFO, a_b.getLevel());
+        assertEquals(Level.INFO, a_b_c.getLevel());
+        // all levels
+        for (final Level level : new Level[] {
+            Level.ALL,
+            Level.CONFIG,
+            Level.FINE,
+            Level.FINER,
+            Level.FINEST,
+            Level.INFO,
+            Level.OFF,
+            Level.SEVERE,
+            Level.WARNING
+        }) {
+            a.setLevel(level);
+            assertEquals(level, a.getLevel());
+            assertTrue(a.isLoggable(level));
+            assertTrue(a_b.isLoggable(level));
+            assertTrue(a_b_c.isLoggable(level));
```

Comment:
The fact that `level` is enabled does not mean that the effective level is `level`. What about a helper functions that return the first of `ALL`,  `FINEST`, `FINER` up to `OFF` that is loggable?

I am also a little bit concerned about the initial conditions: if the only explicitly configured logger is the root logger, shouldn't all `getLevel()` calls return `null` instead of `INFO`?

Reply:
> The fact that level is enabled does not mean that the effective level is level. 

According to the Javadoc, JUL returns null when the level is not set and the effective level is the parent's level, and on up the chain presumably.

> What about a helper functions that return the first of ALL, FINEST, FINER up to OFF that is loggable?

Not useful IMO here since this test goes over all values, and passes, not that this means it's correct ;-) So we should figure this out first before adding more stuff ;-)

> I am also a little bit concerned about the initial conditions: if the only explicitly configured logger is the root logger, shouldn't all getLevel() calls return null instead of INFO?

The new test does not use the root logger though.

Reply:
Well, at least the build is green ;-)